rllib/offline/estimators/direct_method.py

```diff
@@ -0,0 +1,159 @@
+from typing import Tuple, List, Generator
+from ray.rllib.offline.estimators.off_policy_estimator import (
+    OffPolicyEstimator,
+    OffPolicyEstimate,
+)
+from ray.rllib.policy import Policy
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.typing import SampleBatchType
+from ray.rllib.offline.estimators.fqe_torch_model import FQETorchModel
+from ray.rllib.offline.estimators.qreg_torch_model import QRegTorchModel
+from gym.spaces import Discrete
+import numpy as np
+
+torch, nn = try_import_torch()
+
+
+# TODO (rohan): replace with AIR/parallel workers
+# (And find a better name than `should_train`)
+def k_fold_cv(
+    batch: SampleBatchType, k: int, should_train: bool = True
+) -> Generator[Tuple[List[SampleBatch]], None, None]:
+    """Utility function that returns a k-fold cross validation generator
+    over episodes from the given batch. If the number of episodes in the
+    batch is less than `k` or `should_train` is set to False, yields an empty
+    list for train_episodes and all the episodes in test_episodes.
+
+    Args:
+        batch: A SampleBatch of episodes to split
+        k: Number of cross-validation splits
+        should_train: True by default. If False, yield [], [episodes].
+
+    Returns:
+        A tuple with two lists of SampleBatches (train_episodes, test_episodes)
+    """
+    episodes = batch.split_by_episode()
+    n_episodes = len(episodes)
+    if n_episodes < k or not should_train:
+        yield [], episodes
+        return
+    n_fold = n_episodes // k
+    for i in range(k):
+        train_episodes = episodes[: i * n_fold] + episodes[(i + 1) * n_fold :]
+        if i != k - 1:
+            test_episodes = episodes[i * n_fold : (i + 1) * n_fold]
+        else:
+            # Append remaining episodes onto the last test_episodes
+            test_episodes = episodes[i * n_fold :]
+        yield train_episodes, test_episodes
+    return
+
+
+class DirectMethod(OffPolicyEstimator):
+    """The Direct Method estimator.
+
+    q_model_type: Either "fqe" for Fitted Q-Evaluation or "qreg" for Q-Regression, or
+        a custom model that implements `estimate_q(states,actions)`
+        and `estimate_v(states, action_probs)`
+    framework: One of "tf|tf2|tfe|torch", currently only "torch" is supported
+    k: k-fold cross validation for training model and evaluating OPE
+    q_model_kwargs: Optional arguments for the specified Q model
```

Comment:
Although I like this, common you'd find prepended stars, as in `**kwargs`, in other places. So maybe we should keep this uniform.